recipe_caching/caching_query.py

```diff
@@ -102,32 +101,41 @@ def get_value(
         assert not ignore_expiration or not createfunc, \
             "Can't ignore expiration and also provide createfunc"
 
+        ran_createfunc = []
```

Comment:
This is a little janky, but it's the easiest way to do it in Python 2. In python 3, I would say `ran_createfunc = False`, then use `nonlocal` in the inner function which would set it to `True`.